docs/cookbook/create-advanced-grid.md

```diff
@@ -0,0 +1,463 @@
+    {
+-        $grid = $this->salesmanGridFactory->create();
++        $grid = $this->salesmanGridInlineEdit->getGrid();
+
+        return $this->render('@ShopsysShop/Admin/Content/Salesman/list.html.twig', [
+            'gridView' => $grid->createView(),
+        ]);
+    }
+}  
+```
+
+At this point, you should be able to edit and create new salesmen directly in the grid.
+![Advanced grid with inline edit](img/advanced-grid-inline-edit.png)
+
+## 2. Sort data manually (drag and drop)
+In this part, we will enable drag and drop sorting of our salesmen using the grid. To make the changes in the ordering persistent, we need add new attribute to `Salesman` entity, first.
+
+### 2.1 Add `$position` to `Salesman` entity and mark it as DB column using Doctrine ORM annotation
+```diff
+// src/Shopsys/ShopBundle/Model/Salesman/Salesman.php
+
+class Salesman
+{
++    /**
++     * @var int|null
++     *
++     * @ORM\Column(type="integer", nullable=true)
++     */
++    protected $position;
+}
+```
+
+### 2.2 Generate new database migration
+Run phing target
+```bash
+php phing db-migrations-generate
+```
+
+The command prints a file name the migration was generated into.
+The migration will look like this:
+```php
+namespace Shopsys\ShopBundle\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Shopsys\MigrationBundle\Component\Doctrine\Migrations\AbstractMigration;
+
+class Version20190305140005 extends AbstractMigration
+{
+    /**
+     * @param \Doctrine\DBAL\Schema\Schema $schema
+     */
+    public function up(Schema $schema)
+    {
+        $this->sql('ALTER TABLE salesmen ADD position INT DEFAULT NULL');
+    }
+
+    /**
+     * @param \Doctrine\DBAL\Schema\Schema $schema
+     */
+    public function down(Schema $schema)
+    {
+    }
+}
+
+```
+
+### 2.3 Execute migrations to propagate all the changes to the database
+Run phing target
+```bash
+php phing db-migrations
+```
+
+### 2.2 Make the `Salesman` entity implement `OrderableEntityInterface`
+```diff
+// src/Shopsys/ShopBundle/Model/Salesman/Salesman.php
+
++ use Shopsys\FrameworkBundle\Component\Grid\Ordering\OrderableEntityInterface;
+
+- class Salesman
++ class Salesman implements OrderableEntityInterface
+{
++    /**
++     * @param int $position
++     */
++    public function setPosition($position)
++    {
++        $this->position = $position;
++    }
+}
+```
+
+### 2.3 Enable drag and drop sorting in `SalesmanGridFactory`
+```diff
+// src/Shopsys/ShopBundle/Grid/Salesman/SalesmanGridFactory.php
+
+class SalesmanGridFactory implements GridFactoryInterface
+{
+    public function create(): Grid
+    {
+        ...
++       $grid->enableDragAndDrop(Salesman::class);
+        ...
+    {
+}
+```
+
+Now you should be able to sort your salesman using the cross icon in the left part of each row as a handle for drag a drop.
+![Advanced grid with drag and drop](img/advanced-grid-drag-and-drop.png)
+
+## Pitfalls
+- Be aware of using all the combinations that grid provides
```

Comment:
is dash on the start correct?

Reply:
originally, there were more pitfalls :laughing:

Reply:
changed to one sentence